lib/tests/unit/connector/connection_test.cc

```diff
@@ -20,4 +22,84 @@ TEST_CASE("Connection::connect", "[connector]") {
     }
 }
 
+static constexpr int TEST_TIMEOUT = 100;
+
+TEST_CASE("Connection::connect", "[connector]") {
+    SECTION("successfully connects") {
+        ClientMetadata c_m { "test_client", getCaPath(), getCertPath(),
+                             getKeyPath(), TEST_TIMEOUT };
+
+        MockServer mock_server(TEST_TIMEOUT);
+        bool connected = false;
+        mock_server.set_open_handler([&connected](websocketpp::connection_hdl hdl) {
+            connected = true;
+        });
+        mock_server.go();
+
+        Connection connection { "wss://localhost:" + std::to_string(mock_server.port()) + "/pcp", c_m };
+        connection.connect(10);
+        connection.close();
+        REQUIRE(connected);
+    }
+
+    SECTION("successfully connects to failover broker") {
+        ClientMetadata c_m { "test_client", getCaPath(), getCertPath(),
+                             getKeyPath(), TEST_TIMEOUT };
+
+        MockServer mock_server(TEST_TIMEOUT);
+        bool connected = false;
+        mock_server.set_open_handler([&connected](websocketpp::connection_hdl hdl) {
+            connected = true;
+        });
+        mock_server.go();
+
+        auto port = mock_server.port();
+        Connection connection { std::vector<std::string> { "wss://localhost:" + std::to_string(port+1) + "/pcp",
+                                                           "wss://localhost:" + std::to_string(port) + "/pcp" }, c_m };
+        connection.connect(30);
+        connection.close();
+        REQUIRE(connected);
+    }
+
+    SECTION("successfully connects to failover when primary broker disappears") {
+        ClientMetadata c_m { "test_client", getCaPath(), getCertPath(),
+                             getKeyPath(), TEST_TIMEOUT };
+
+        bool connected_a = false, connected_b = false, connected_c = false;
+
+        std::unique_ptr<MockServer> mock_server_a(new MockServer(TEST_TIMEOUT));
+        mock_server_a->set_open_handler([&connected_a](websocketpp::connection_hdl hdl) {
+            connected_a = true;
+        });
+
+        std::unique_ptr<MockServer> mock_server_b(new MockServer(TEST_TIMEOUT));
+        mock_server_b->set_open_handler([&connected_b](websocketpp::connection_hdl hdl) {
+            connected_b = true;
+        });
+
+        auto port_a = mock_server_a->port();
+        auto port_b = mock_server_b->port();
+        Connection connection { std::vector<std::string> { "wss://localhost:" + std::to_string(port_a) + "/pcp",
+                                                           "wss://localhost:" + std::to_string(port_b) + "/pcp" }, c_m };
+
+        mock_server_a->go();
+        connection.connect(10);
+        REQUIRE(connected_a);
+        mock_server_a.reset();
+
```

Comment:
We should ensure that the Connection instance gets closed here, before reconnecting. This should work:
`
static void let_connection_stop(Connection& connection, int timeout = 2)
{
    lth_util::Timer timer {};
    while (connection.getConnectionState() == ConnectionState::open
            && timer.elapsed_seconds() < timeout)
        Util::this_thread::sleep_for(Util::chrono::milliseconds(1));
    REQUIRE(connection.getConnectionState() != ConnectionState::open);
}
`